### ISSUE TITLE
Must field values be non objects/arrays?

### ISSUE THREAD

**Issue by webmasterkai** [erikras/redux-form#379]
This might be a duplicate of #360 and/or #365.

Is it possible to have a field with an object value? It's not deep or nested. I just want to be able to have a field with an object as a value. It works correctly when I return an object on `onChange()`. The value I get when the form is submitted is an object. But it does not work when I have the value in `initialValues`.

The following test works where the `photo` field has an object value. Indicating that it should be possible?

```
describe('getValues', () => {
  it('should get values from form', () => {
    const form = {
      foo: {value: 'bar'},
      catLives: {value: 9},
      alive: {value: true},
      nullValue: null,
      photo: {value: {alt: 'text', url: '//example.com/img.jpg'}}
    };
    const fields = ['foo', 'catLives', 'alive', 'nullValue', 'photo'];
    expect(getValues(fields, form))
      .toBeA('object')
      .toEqual({
        foo: 'bar',
        catLives: 9,
        alive: true,
        nullValue: null,
        photo: {alt: 'text', url: '//example.com/img.jpg'}
      });
  });
});
```
But it looks to me like there is a potentially conflicting test intended for deep forms.
https://github.com/erikras/redux-form/blob/master/src/__tests__/initializeState.spec.js#L32-L61

Thanks!

**Comment by webmasterkai** [comment]
I'm not sure what was wrong with my `initialValues` but I figured out how to get it to work like I wanted.

**Comment by dlmanning** [comment]
@webmasterkai : Could I ask how?

**Comment by webmasterkai** [comment]
It's possible my code wasn't correctly checking for `defaultValue` and was only looking at `value`. Other than that I'm not entirely sure.

**Comment by erikras** [comment]
Yes, you _should_ be able to have an object or an array as your value, but that is broken at the moment. #360.

**Comment by skyblue** [comment]
same problem~~ after update to v3.1.7

**Comment by erikras** [comment]
This should be fixed in [`v4.0.0`](https://github.com/erikras/redux-form/releases/tag/v4.0.0). Come and get it!
